### PR TITLE
Add wait no initializing shards

### LINKED ISSUE #25623: Add a wait for no intializing shards parameter to cluster health
Sometime it's handy to know everything has stabilized in the cluster when we run rest tests on it. A common problem is that we run the test on varying number of nodes. This means we sometime have assigned replicas other times there is a single node and thus the replicas will stay unassigned. This means we can't use wait for green or wait for yellow. 

It will be handy to add a new options to the cluster health requests allowing to wait for all shard initialization to settle down (similar to the current option to wait for all relocations to stop). The new option may even supersede the relocation option (as it is a super set).

### PR DESCRIPTION
**Pull request by matarrese**
Hello, I don't know if @Hogusong is still working on this issue, so I decided to bring it up and send you a pull request. It is not intended to be ready to be tested, as it is failing some tests. I am just asking if it's possible to have a review and understand if it goes in the right direction. If other people working on this, please feel free to close it.
Closes #25623

### CODE REVIEW

**Comment by ywelsch** [comment]
Sure, this looks good so far. It still needs adaptions on the REST layer, tests and all that.

File: core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java [hunk 1]
```diff
@@ -221,6 +224,10 @@ private boolean prepareResponse(final ClusterHealthRequest request, final Cluste
         if (request.waitForNoRelocatingShards() && response.getRelocatingShards() == 0) {
             waitForCounter++;
         }
+        if (request.waitForNoInitializingShards() && !clusterState.getRoutingNodes().hasInactivePrimaries() &&
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java]: you can use the same approach as for the `waitForNoRelocatingShards` a few lines above, i.e. `request.waitForNoInitializingShards() && response.getInitializingShards() == 0`

**Comment by ywelsch** [comment]
@matarrese any updates on this? I'll close the PR due to inactivity. Please let us know if you still want to work on this.

**Comment by matarrese** [comment]
Hi @ywelsch , happy to go back working on this. Is it possible to open the pull request?

**Comment by jasontedor** [comment]
This was addressed in #27489.

**Pull request closed without merging**